src/CMakeLists.txt

```diff
@@ -271,20 +240,133 @@ set (NN_SOURCES
 )
 
 if (WIN32)
-    LIST (APPEND NN_SOURCES
```

Comment:
> it will be easy for folks to think they're modifying the file list for all platforms, and only be modifying them for one of them

This specifically is a weak argument, but I'm generally OK with removing these changes in favor of a later, more sustainable rearchitecture.

Keep in mind though -- there *are* functional changes in the *.h and *.c files in this PR -- notably, the order of precedence into which `#if` a given system would fall. And the functional change within `nn.h` is simply to support a previously unsupported feature of CMake, and that is building nanomsg as a static lib on Windows.